pootle/apps/pootle_app/management/commands/refresh_scores.py

```diff
@@ -28,10 +29,51 @@ def add_arguments(self, parser):
             default=False,
             help='Reset all scores to zero',
         )
+        parser.add_argument(
+            '--user',
+            action='append',
+            dest='users',
+            help='User to refresh',
+        )
 
     def handle(self, **options):
+        self.stdout.write('Start running of refresh_scores command...')
+
+        User = get_user_model()
+        users = User.objects.all()
+        if options['users']:
+            users = users.filter(username__in=options['users'])
 
         if options['reset']:
-            User = get_user_model()
-            User.objects.all().update(score=0)
-            ScoreLog.objects.all().delete()
+            users.update(score=0)
+            scorelogs = ScoreLog.objects.all()
+            if options['users']:
+                scorelogs = scorelogs.filter(user__in=users)
+
+            scorelogs.delete()
+
+            if options['users']:
+                self.stdout.write('Scores for specified users were reset to 0.')
+            else:
+                self.stdout.write('Scores for all users were reset to 0.')
+            return
+
+        start = datetime.datetime.now()
+        for user_pk, username in users.values_list("pk", "username"):
+            self.stdout.write("Processing user %s..." % username)
+            scorelog_qs = ScoreLog.objects.filter(user=user_pk) \
+                .select_related(
+                    'submission',
+                    'submission__suggestion',
+                    'submission__unit')
+            user_score = 0
+            for scorelog in scorelog_qs.iterator():
+                score_delta = scorelog.get_score_delta()
+                user_score += score_delta
+                ScoreLog.objects.filter(id=scorelog.id) \
+                    .update(score_delta=score_delta)
+            self.stdout.write("Set user score for %s to %.3f" %
```

Comment:
I would rephrase this to align with previous messages: `Score for user %s set to %.3f`